src/infra/flow-system/sqlite/src/flow_system_event_store_sqlite.rs

```diff
@@ -0,0 +1,300 @@
+                r#"
+INSERT INTO flow_configuration_event(created_time)
+"#,
+            );
+
+            let created_times = vec![Utc::now(); events.len()];
+
+            query_builder.push_values(created_times, |mut b, created_time| {
+                b.push_bind(created_time);
+            });
+
+            query_builder.push("RETURNING event_id");
+
+            query_builder
+                .build_query_as::<ReturningEventModel>()
+                .fetch_all(connection_mut)
+                .await
+                .int_err()?
+        };
+
+        let connection_mut = tr.connection_mut().await?;
+        let mut query_builder = match flow_key {
+            FlowKey::Dataset(fk_dataset) => {
+                let mut query_builder = QueryBuilder::<Sqlite>::new(
+                    r#"
+INSERT INTO dataset_flow_configuration_events (event_id, dataset_id, dataset_flow_type, event_type, event_time, event_payload)
+"#,
+                );
+
+                query_builder.push_values(
+                    events.into_iter().zip(flow_configuration_events),
+                    |mut b, (event, ReturningEventModel { event_id })| {
+                        b.push_bind(event_id);
+                        b.push_bind(fk_dataset.dataset_id.to_string());
+                        b.push_bind(fk_dataset.flow_type);
+                        b.push_bind(event.typename());
+                        b.push_bind(event.event_time());
+                        b.push_bind(serde_json::to_value(event).unwrap());
+                    },
+                );
+
+                query_builder
+            }
+            FlowKey::System(fk_system) => {
+                let mut query_builder = QueryBuilder::<Sqlite>::new(
+                    r#"
+INSERT INTO system_flow_configuration_events (event_id, system_flow_type, event_type, event_time, event_payload)
+"#,
+                );
+
+                query_builder.push_values(
+                    events.into_iter().zip(flow_configuration_events),
+                    |mut b, (event, ReturningEventModel { event_id })| {
+                        b.push_bind(event_id);
+                        b.push_bind(fk_system.flow_type);
+                        b.push_bind(event.typename());
+                        b.push_bind(event.event_time());
+                        b.push_bind(serde_json::to_value(event).unwrap());
+                    },
+                );
+
+                query_builder
+            }
+        };
+
+        query_builder.push("RETURNING event_id");
+
+        let rows = query_builder
+            .build_query_as::<ReturningEventModel>()
+            .fetch_all(connection_mut)
+            .await
+            .int_err()?;
+
+        if let Some(last_row) = rows.last() {
+            Ok(EventID::new(last_row.event_id))
+        } else {
+            let event_id = i64::try_from(self.len().await.int_err()?).int_err()?;
+
+            Ok(EventID::new(event_id))
+        }
```

Comment:
Same here, let's experiment if we really need this condition

Reply:
Removed -- added a check at the method contract level

Done